cirq-core/cirq/ops/linear_combinations.py

```diff
@@ -839,10 +839,13 @@ def __pow__(self, exponent: int):
         if exponent == 0:
             return PauliSum(value.LinearDict({frozenset(): 1 + 0j}))
         if exponent > 0:
-            base = self.copy()
-            for _ in range(exponent - 1):
-                base *= base
-            return base
+            remainder = PauliSum.from_pauli_strings([PauliString(coefficient=1)])
+            while exponent > 0 :
+                if exponent & 1 :
+                    remainder = remainder * self
+                self = self * self
```

Comment:
You should not change `self`. Instead make a copy of `self` and use that for exponentiation.

Reply:
thanks for the review, I have modified the code to use a copy.

Reply:
Added tests as well on the latest commit